src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -116,10 +122,15 @@ private bool CheckGlobalConditions(Log log, DateTime lastSuccessfulBuildStartTim
                 return log.Fail("Disabled", nameof(Resources.FUTD_DisableFastUpToDateCheckTrue));
             }
 
-            if (validateFirstRun && !state.WasStateRestored && lastSuccessfulBuildStartTimeUtc == DateTime.MinValue)
+            if (validateFirstRun && lastSuccessfulBuildStartTimeUtc is null)
```

Comment:
`WasStateRestored` has been removed. It's enough to check whether `lastSuccessfulBuildStartTimeUtc` is `null` or not.